quality_check/07-technology_diff_exp.Rmd

```diff
@@ -0,0 +1,189 @@
+This will be captured by `sessionInfo()` below.
+
+## Set up
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+```{r}
+library(limma)
+library(GenomicRanges)
+library(rtracklayer)
+library(Rsamtools)
+```
+
+```{r}
+results_dir <- "results"
+dir.create(results_dir, showWarnings = FALSE)
+```
+
+## Read in data
+
+Test compendium expression matrix
+
+```{r}
+exprs_mat_file <- file.path("data", "larger_compendium_exprs_mat.RDS")
+exprs_mat <- readRDS(exprs_mat_file)
+```
+
+Generate technology labels using the sample names
+
+```{r}
+sample_names <- colnames(exprs_mat)
+compendium_technology_labels <- rep("MICROARRAY", length(sample_names))
+compendium_technology_labels[grep("SRR|ERR|DRR", sample_names)] <- "RNASEQ"
+```
+
+## Differential expression
+
+Test for genes that are differentially expressed between technologies
+
+```{r}
+design_matrix <- model.matrix(~ compendium_technology_labels)
+fit <- lmFit(exprs_mat, design = design_matrix)
+fit2 <- eBayes(fit)
+diff_exp_stats <- topTable(fit2, number = nrow(exprs_mat)) %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("Gene")
+```
+
+Save to file
+
+```{r}
+readr::write_tsv(diff_exp_stats, 
+                 file.path(results_dir, 
+                           "larger_compendium_diff_exprs_technologies.tsv"))
+```
+
+Let's take a peak at the "top" differentially expressed genes.
+
+```{r}
+diff_exp_stats %>%
+  dplyr::filter(adj.P.Val < 0.00001) %>%
+  dplyr::arrange(`t`) %>%
+  head()
+```
+
+```{r}
+data.frame(
+  expression_value = exprs_mat["ENSDARG00000098608", ],
+  technology = compendium_technology_labels
+) %>%
+  ggplot2::ggplot(ggplot2::aes(x = technology, y = expression_value)) +
+  ggplot2::geom_boxplot() +
+  ggplot2::labs(title = "ENSDARG00000098608", subtitle = "t = -234.7375") +
+  ggplot2::theme_bw()
+```
+
+Genes with negative `t` values have lower values in RNA-seq samples.
+
+And the genes that have higher values in RNA-seq samples (positive `t`).
+
+```{r}
+diff_exp_stats %>%
+  dplyr::filter(adj.P.Val < 0.00001) %>%
+  dplyr::arrange(`t`) %>%
+  tail()
+```
+
+```{r}
+data.frame(
+  expression_value = exprs_mat["ENSDARG00000060767", ],
+  technology = compendium_technology_labels
+) %>%
+  ggplot2::ggplot(ggplot2::aes(x = technology, y = expression_value)) +
+  ggplot2::geom_boxplot() +
+  ggplot2::labs(title = "ENSDARG00000060767", subtitle = "t = 165.2244") +
+  ggplot2::theme_bw()
+```
+
+
+### Relationship with gene lengths?
+
+Is there any relationship between the length of a gene and it's `t` statistic?
+
+`GRCz11` related files were obtained in `00-data_download.sh`
+
+Adapted from:
+https://github.com/dpryan79/Answers/blob/d2d7366fec907fd92ccf03e77c2d53cfe45b02c9/SEQanswers_42420/GTF2LengthGC.R
+
+See also: https://bioinformatics.stackexchange.com/questions/2567/how-can-i-calculate-gene-length-for-rpkm-calculation-from-counts-data
+
+```{r}
+gtf_file <- file.path("data", "GRCz11", "Danio_rerio.GRCz11.95.gtf")
```

Comment:
Sometimes I like declaring all these file paths toward the beginning of the file in case they need to change, but also I can also see the strategy of putting it right before it's being used. So meh. IDK.